exec/java-exec/src/main/java/org/apache/drill/exec/ops/QueryContext.java

```diff
@@ -273,6 +279,29 @@ public RemoteFunctionRegistry getRemoteFunctionRegistry() {
     return drillbitContext.getRemoteFunctionRegistry();
   }
 
+  /**
+   * Check if auto-limiting of resultset is enabled
+   * @return True if auto-limit is enabled
+   */
+  public boolean isAutoLimitEnabled() {
+    return autoLimitRowCount != null;
+  }
+
+  /**
+   * Returns the maximum size of auto-limited resultset
+   * @return Maximum size of auto-limited resultSet
+   */
+  public Integer getAutoLimitRowCount() {
```

Comment:
The feature is specifically for WebUI submitted queries. A query option doesn't make sense because WebUI queries are "session-less" and the only reason we want to wrap with a limit is to prevent the Drillbit from being overloaded with holding an entire result set that does not get completely consumed by the HTTP client.

Reply:
@kkhatua , I think adding new query scope option isn't a bad idea, because now same ```autoLimitRowCount``` and related methods are spread across lots of places, and if it's possible to minimize the duplication let's give it a try. Also options can have good long descriptions where you can mention all details about how it's used and  why it was added.

Reply:
Done in the latest set of commits